admin_guide/downgrade.adoc

```diff
@@ -0,0 +1,249 @@
+
+== Step 1: Verify Backups Are In Place
+
+The openshift-ansible upgrade playbooks should have created a backup of the master-config.yaml, and the etcd data directory. Please ensure these exist on your masters/etcd members. In the case of a separate etcd cluster, the backup is likely created on all etcd members, though we'll only need one to recover.
+
+====
+----
+/etc/openshift/master/master-config.yaml.[TIMESTAMP]
+/var/lib/openshift/etcd-backup-[TIMESTAMP]
+----
+====
+
+The rpm downgrade will likely create .rpmsave backups of these files, but it may be a good idea to keep a separate copy of them regardless:
+
+====
+----
+/etc/sysconfig/openshift-master
+/etc/etcd/etcd.conf (if using a separate etcd cluster)
+----
+====
+
+
+== Step 2: Shutdown Cluster
+
+On all masters, nodes, and etcd members (if using a separate etcd cluster), ensure relevant services are stopped.
+
+====
+----
+$ systemctl stop atomic-openshift-master
+$ systemctl stop atomic-openshift-node
+$ systemctl stop etcd
+----
+====
+
+== Step 3: Remove 3.1 Atomic OpenShift RPMs
+
+On each master / node / etcd member:
+
+====
+----
+yum remove atomic-openshift atomic-openshift-clients atomic-openshift-node atomic-openshift-master etcd openvswitch tuned-projects-atomic-openshift-node atomic-openshift-sdn-ovs tuned-profiles-atomic-openshift-node
+----
+====
+
+
+== Step 4: Re-install 3.0 RPMs
+
+Disable 3.1 repositories, and re-enable 3.0 repositories:
+
+====
+----
+$ subscription-manager repos --disable=rhel-7-server-ose-3.1-rpms --enable=rhel-7-server-ose-3.0-rpms
+----
+====
+
+On each OpenShift master:
+
+====
+----
+$ yum install openshift openshift-master openshift-node openshift-sdn-ovs
+----
+====
+
+On each OpenShift node:
+
+====
+----
+$ yum install openshift openshift-node openshift-sdn-ovs
+----
+====
+
+If using a separate etcd cluster, on each etcd member:
+
+====
+----
+$ yum install etcd
+----
+====
+
+
+== Stage 5: Restore etcd
+
+=== Create New Etcd Cluster From Backup
+
+For both embedded etcd as well as separate etcd clusters, the first step is to restore the backup by creating a new single node etcd cluster.
+
+Choose a system to be the initial etcd member and restore it's backup and configuration.
+
+WARNING: If you are using embedded non-clustered etcd, please use /var/lib/openshift/openshift.local.etcd for ETCD_DIR in the commands below. If you are using a separate etcd cluster, please use /var/lib/etcd/ for ETCD_DIR.
+
+====
+----
+$ mv $ETCD_DIR /var/lib/etcd.orig
+$ cp -Rp /var/lib/openshift/etcd-backup-20151120093517/ $ETCD_DIR
+$ chcon -R --reference /var/lib/etcd.orig/ $ETCD_DIR
+$ chown -R etcd:etcd $ETCD_DIR
+----
+====
+
+If you are using a separate etcd cluster, you should also restore /etcd/etcd/etcd.conf from backup or .rpmsave.
```

Comment:
Drive-by comment (will review doc more fully ASAP):

Think this should be /etc/etcd/etcd.conf (has an extra "d" currently).